### PR TITLE
add app insight BE

### PR DESCRIPTION
**Pull request by robertstriz**
## Summary

🚨 Provide a brief summary of your work.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactoring (non-breaking change to code and structure)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Testing

🚨 Provide info about the new tests and tests result

## Notes

🚨 Anything else

### CODE REVIEW

File: src/WebApi/Logging/TimeElapsedDiagnosticsMiddleware.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        try
+        {
+            await _next(context);
+            PushPropertyElapsed(sw);
+            Log.Debug("DiagnosticsMiddleware:OK");
+        }
+        catch (Exception e)
+        {
+            PushPropertyElapsed(sw);
+            Log.Fatal("DiagnosticsMiddleware:ERROR Message: {message} , Exception: {exception}", e.Message,e);
```
Comment (fpindej) [inline src/WebApi/Logging/TimeElapsedDiagnosticsMiddleware.cs:29]:
```suggestion
            Log.Fatal("TimeElapsedDiagnosticsMiddleware:ERROR Message: {message} , Exception: {exception}", e.Message,e);
```

**Review by fpindej — CHANGES_REQUESTED** [review]
This looks good, it works as well. Resolve my comments and it's a go for me 😊 🥳

**Comment by robertstriz** [comment]
Comments resolved

**Review by fpindej — DISMISSED** [review]
Good to go

**Comment by fpindej** [comment]
@PlesnikJakub Could you please check the pipeline? Looks like it has some running issues, it's been failing and I'm sure this project is fine. At least it works on both @robertstriz and my machines.

**Comment by PlesnikJakub** [comment]
> @PlesnikJakub Could you please check the pipeline? Looks like it has some running issues, it's been failing and I'm sure this project is fine. At least it works on both @robertstriz and my machines.

I can't tell why but the solution file doesn't seem to be right.
```
D:\a\interview-app-api\interview-app-api\src\WebApi\WebApi.sln : Solution file error MSB5009: Error parsing the project "Logging" section with GUID: "{532A4DCF-02E6-4D72-971E-3E2EB6DA6F24}". It is nested under "{BD63581C-1EC5-4102-94E0-F1003CECE2D9}" but that project is not found in the solution.
```

 I propose to change the ID before the Loggin project to the same ID as with the other projects 
```
- Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Logging", "Logging\Logging.csproj", "{532A4DCF-02E6-4D72-971E-3E2EB6DA6F24}"
+ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Logging", "Logging\Logging.csproj", "{532A4DCF-02E6-4D72-971E-3E2EB6DA6F24}"
```

and remove the section about nested project

```
GlobalSection(NestedProjects) = preSolution
        {532A4DCF-02E6-4D72-971E-3E2EB6DA6F24} = {BD63581C-1EC5-4102-94E0-F1003CECE2D9}
EndGlobalSection
```

**Comment by PlesnikJakub** [comment]
One more thing, I see `ConfigurationExtension` but I feel there may be some overlap with `ConfigurationProvider`.
Using extension means we won't be able to use this in any other project than WebApi, that ofc may or may not be a problem.

Let's just consider if it would make sense to merge those implementations together.

**Review by PlesnikJakub — APPROVED** [review]

**Pull request merged**